chain/validation/state.go

```diff
@@ -37,11 +43,25 @@ func NewState() *StateWrapper {
 		panic(err) // Never returns error, the error return should be removed.
 	}
 	storageImpl := &directStorage{cst}
-	return &StateWrapper{bs, newKeyStore(), treeImpl, storageImpl}
+
+	sr, err := treeImpl.Flush()
+	if err != nil {
+		panic(err)
+	}
+	return &StateWrapper{bs, newKeyStore(), treeImpl, storageImpl, sr}
 }
 
 func (s *StateWrapper) Cid() cid.Cid {
-	panic("implement me")
+	return s.stateRoot
+}
+
+// TODO very unsure about this, corrently its the easiest way to handle updating the statewrappers CID.
+func (s *StateWrapper) updateStateRoot() {
```

Comment:
Need to call after setting an actor on the state tree, open to better options here.

Reply:
I think this is the right general approach, but I made it easier to get right by removing the state tree reference from the state wrapper, storing only the CID.